File: css-validator/javacc/examples/JJTreeExamples/cpp/ASTMyOtherID.cc
/*
 * ASTMyOtherID.cpp
 *
 *  Created on: 27 mars 2014
 *      Author: FrancisANDRE
 */

#include "ASTMyOtherID.h"
#include "ParserVisitor.h"

namespace @NAMESPACE@ {

ASTMyOtherID::ASTMyOtherID(int id) : SimpleNode(id) {
}
ASTMyOtherID::ASTMyOtherID(Parser *p, int id) : SimpleNode(p, id){
}
ASTMyOtherID::~ASTMyOtherID() {
}

void
ASTMyOtherID::setName(JAVACC_STRING_TYPE image) {
	name = image;
}


JAVACC_STRING_TYPE
ASTMyOtherID::toString() const {
  return "Identifier: " + name;
}
/** Accept the visitor. **/
void*
ASTMyOtherID::jjtAccept(ParserVisitor *visitor, void * data) const {
  return visitor->visit(this, data);
}


} /* namespace @NAMESPACE@ */

File: css-validator/javacc/examples/JJTreeExamples/java/ASTMyOtherID.java
public
class ASTMyOtherID extends SimpleNode {
  private String name;
  public ASTMyOtherID(int id) {
    super(id);
  }

  public ASTMyOtherID(Eg4 p, int id) {
    super(p, id);
  }

  /**
   * Set the name.
   * @param n the name
   */
  public void setName(String n) {
    name = n;
  }

  /**
   * {@inheritDoc}
   * @see org.javacc.examples.jjtree.eg2.SimpleNode#toString()
   */
  public String toString() {
    return "Identifier: " + name;
  }

  /** Accept the visitor. **/
  public Object jjtAccept(Eg4Visitor visitor, Object data) {
    return visitor.visit(this, data);
  }
}
